lib/cn/kvset_split.h

```diff
@@ -36,17 +36,30 @@ struct kvset_split_res {
 };
 
 /**
- * kvset_split() - split kblocks, vblocks and the hblock of a given kvset
+ * struct kvset_split_wargs - split work args
+ *
+ * @work:       work struct
  * @ks:         kvset handle
  * @split_kobj: split key object
  * @pc:         perfc_set handle
+ * @err:        return status of this split work
  * @result:     split result (output)
  */
-merr_t
-kvset_split(
-    struct kvset           *ks,
-    const struct key_obj   *split_kobj,
-    struct perfc_set       *pc,
-    struct kvset_split_res *result);
+struct kvset_split_wargs {
+    struct work_struct     work;
+    struct kvset          *ks;
+    struct key_obj        *split_kobj;
+    struct perfc_set      *pc;
+    merr_t                 err;
+    struct kvset_split_res result;
+};
+
+/**
+ * kvset_split_worker() - worker function that splits a given kvset
```

Comment:
I think we should at least move away from this comment pattern where the name of the struct/function is repeated.

Reply:
Wasn't this a requirement of some tool that parses these comments and generates docs? I don't know.

Reply:
Whatever tool that was, I don't think it is around anymore. Not even doxygen works this way in the public headers.

Reply:
I thought it was required by doxygen, right?  We went through this time-consuming exercise to document all functions with this syntax.  Not only was the information far away from the actual code that it referenced, it would often dward the function in lines of code.   On top of that, it would go stale...

Reply:
Nope, not required by doxygen. It parses names automatically.

Reply:
https://doxygen.nl/manual/docblocks.html
The docs seems different than when I looked at them years ago, but they seem to have all sorts of annotations for function name and what not...  Confusing..